### PR TITLE
docs(ref): Link to docs.rs metadata table

### PR DESCRIPTION
**Pull request by epage**
This gets us part of the cross-linking described in #7164

### CODE REVIEW

File: src/doc/src/reference/manifest.md [hunk 1]
```diff
@@ -511,6 +511,9 @@ package-name = "my-awesome-android-app"
 assets = "path/to/static"
 ```
 
+First-party `package.metadata` tables:
```
Comment (weihanglo) [inline src/doc/src/reference/manifest.md:514]: Feel a bit abrupt we suddenly mention this. Not a good one but something like below?

```suggestion
Here is a list of `package.metadata` definitions maintained by the Rust Project:
```

Comment (epage) [reply src/doc/src/reference/manifest.md]: Did a bit of a hybrid.

> You'll need to look in the documentation for your tool to see how to use this field.
> For Rust Projects that use `package.metadata` tables, see:
> - [docs.rs](https://docs.rs/about/metadata)

Thoughts?

Comment (weihanglo) [reply src/doc/src/reference/manifest.md]: Way better than mine!

**Review by weihanglo — APPROVED** [review]

**Pull request merged**